src/kixi/stats/distribution.cljc

```diff
@@ -654,30 +661,38 @@
   "Returns a beta distribution.
   Params: {:n ∈ ℕ > 0, :alpha ∈ ℝ > 0, :beta ∈ ℝ > 0}"
   [{:keys [n alpha beta] :or {alpha 1.0 beta 1.0}}]
+  (assert (pos-int? n) (str "n (" n ") must be a positive integer."))
+  (assert (and (pos? alpha) (pos? beta)) (str "alpha (" alpha ") and beta (" beta ") must be positive."))
   (->BetaBinomial n alpha beta))
 
 (defn weibull
   "Returns a weibull distribution.
   Params: {:shape ∈ ℝ >= 0, :scale ∈ ℝ >= 0}"
   [{:keys [shape scale] :or {shape 1.0 scale 1.0}}]
+  (assert (and (not (neg? shape)) (not (neg? scale)))
```

Comment:
Perhaps a `non-neg?` helper function would help tidy up some of these asserts?

Reply:
Okay. Added that to the top of the file. And used it.